src/downloader.rs

```diff
@@ -61,33 +59,29 @@ impl RoutingMessageHandler for GossipRouter {
 
 		counter.channel_announcements += 1;
 		let gossip_message = GossipMessage::ChannelAnnouncement(msg.clone());
-		let detected_gossip_message = DetectedGossipMessage {
-			message: gossip_message,
-			timestamp_seen: timestamp_seen as u32,
-		};
-		let sender = self.sender.clone();
-		tokio::spawn(async move {
-			let _ = sender.send(detected_gossip_message).await;
-		});
+		if let Err(err) = self.sender.try_send(gossip_message) {
+			let gossip_message = match err { TrySendError::Full(msg)|TrySendError::Closed(msg) => msg };
```

Comment:
why are you re-extracting and redeclaring the gossip message here? Is it worth all the extra text to avoid a clone?